File: app/src/main/java/yagoc/players/Player.java
package yagoc.players;

import yagoc.board.BoardReader;
import yagoc.board.Move;
import yagoc.pieces.PieceColor;

public interface Player {
    Move move(BoardReader board);

    PlayerType type();

    PieceColor pieceColor();

    String name();

    default boolean isUser() {
        return type() == PlayerType.user;
    }

    default boolean isComputer() {
        return type() == PlayerType.computer;
    }
}

File: app/src/main/kotlin/yagoc/players/Player.kt
package yagoc.players

import yagoc.board.BoardView
import yagoc.board.Move
import yagoc.pieces.PieceColor

interface Player {
    fun move(board: BoardView): Move
    fun type(): PlayerType
    fun pieceColor(): PieceColor
    fun name(): String
    val isUser: Boolean
        get() = type() == PlayerType.User
    val isComputer: Boolean
        get() = type() == PlayerType.Computer
}